src/Components/Header/Header.tsx

```diff
@@ -32,6 +40,34 @@ const PrimaryHeader = () => {
 
 const SecondaryHeader = () => {
   const location = useLocation();
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+
+  const [showWarningPopup, setShowWarningPopup] = useState(false);
+  const isTimestamped = useAppSelector(selectTimestampStatus);
+  const signatureDownloaded = useAppSelector(selectDownloadStatus);
```

Comment:
This will also be true if I download the signature before adding the timestamp.

Reply:
Yes, you are right. How about checking timestamp status first and then setting downloaded to true? 
`  if (isTimestamped) dispatch(updateDownloadStatus(true)); `

Reply:
Yeah that should work